src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -15,15 +18,24 @@ function deduplicate<T>(someArray: (T | undefined | null)[]) {
   return _.compact(Array.from(new Set(someArray)));
 }
 
-function subclassDefFromLoadout(loadout: Loadout, d2Definitions: D2ManifestDefinitions) {
+function subclassFromLoadout(
+  loadout: Loadout,
+  d2Definitions: D2ManifestDefinitions,
+  allItems: DimItem[] | undefined,
+  store: DimStore | undefined,
+) {
   for (const item of loadout.items) {
-    const itemDef = d2Definitions?.InventoryItem.get(item.hash);
-    if (itemDef?.itemType === DestinyItemType.Subclass) {
-      return itemDef;
+    const resolvedItem = findItemForLoadout(d2Definitions, allItems || [], store?.id, item);
+    if (resolvedItem?.bucket.hash === BucketHashes.Subclass) {
+      return resolvedItem;
```

Comment:
Swapped this over to using the proper resolve item method which @robojumper indicated is more correct for subclasses as they can be "upgraded" to a newer version.